### ISSUE TITLE
Printing VectorFloat3 of size <4 throws IllegalFormatConversionException

### ISSUE THREAD

**Issue by rasmysamy** [beehive-lab/TornadoVM#60]
**Describe the bug**
Printing VectorFloat3 of size <4 throws IllegalFormatConversionException

**How To Reproduce**
Attempt to print a VectorFloat3 of size less than four, for example :
```
VectorFloat3 circleCenters = new VectorFloat3(2);

circleCenters.set(0, new Float3(0,0,10));
circleCenters.set(1, new Float3(1000,1000,2000));

System.out.println(circleCenters);
```

**Expected behavior**

Output:
`<[0, 0, 10], [1000, 1000, 2000]>`

**Computing system setup (please complete the following information):**
 - OS: Arch Linux
 - OpenCL Version 1.2
 - TornadoVM commit id 29b4554

**Additional context**

This is caused by the following code in uk.ac.manchester.tornado.api.collections.types.VectorFloat3:
```
public String toString() {
        return this.numElements > 3 ? String.format("VectorFloat3 <%d>", this.numElements) : this.toString("{%.3f,%.3f,%.3f}");
    }
```

Which fails as the elements are of type Float3 and not float. Instead, it should be:

```
public String toString() {
        if (this.numElements > 3)
            return String.format("VectorFloat3 <%d>", this.numElements);
        String tempString = "";
        for (int i = 0; i<numElements; i++){
            tempString += (" " + this.get(i).toString)
        }
        return tempString
    }
```

**Comment by stratika** [comment]
That seems a good point. It seems that the `Float3` object is passed internally in the String Formatter class which fails to convert it to a sequence of floats and throws the exception. 

```
Exception in thread "main" java.util.IllegalFormatConversionException: f != uk.ac.manchester.tornado.api.collections.types.Float3
	at java.util.Formatter$FormatSpecifier.failConversion(Formatter.java:4302)
	at java.util.Formatter$FormatSpecifier.printFloat(Formatter.java:2806)
	at java.util.Formatter$FormatSpecifier.print(Formatter.java:2753)
	at java.util.Formatter.format(Formatter.java:2520)
	at java.util.Formatter.format(Formatter.java:2455)
	at java.lang.String.format(String.java:2940)
	at uk.ac.manchester.tornado.api.collections.types.VectorFloat3.toString(VectorFloat3.java:175)
	at uk.ac.manchester.tornado.api.collections.types.VectorFloat3.toString(VectorFloat3.java:193)
	at uk.ac.manchester.tornado.examples.vectors.VectorAddTest.main(VectorAddTest.java:41)
```

We will fix it soon. Thanks for spotting it.

**Comment by mikepapadim** [comment]
I am looking into this issue.

**Comment by jjfumero** [comment]
@mikepapadim , coordinate with @stratika . He is solving this issue as well.

**Comment by mikepapadim** [comment]
Hello @rasmysamy, thank you for your feedback.
Your fix will be upstream through PR #68.

**Comment by mikepapadim** [comment]
This is now fixed with #68. 
I think we can close it.
